app/controllers/user_controller.rb

```diff
@@ -0,0 +1,28 @@
+class UserController < ApplicationController
+  def new
+    layout 'home'
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      sign_in @user
+      redirect_to user_path(@user)
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @prev_events = current_user.attended_event.previous_events
+    @upcoming_events = current_user.attended_event.upcoming_events
```

Comment:
Not for the current user but for the user which the `id` in the `params` that is `@user`

Reply:
Done.